src/goSuggest.ts

```diff
@@ -214,7 +215,12 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 	}
 	// TODO: Shouldn't lib-path also be set?
 	private ensureGoCodeConfigured(): Thenable<void> {
-		let setPkgsList = getImportablePackages(vscode.window.activeTextEditor.document.fileName).then(pkgMap => {
+		let importablePkgsPromise: Promise<Map<string, string>> = getImportablePackages(vscode.window.activeTextEditor.document.fileName);
+		if (this.pkgsList.size) {
+			importablePkgsPromise = Promise.race([timeout(1000).then(() => this.pkgsList), importablePkgsPromise]);
+		}
+
+		let setPkgsList = importablePkgsPromise.then(pkgMap => {
 			this.pkgsList = pkgMap;
```

Comment:
How about this?
```
let importablePkgsPromise = getImportablePackages(vscode.window.activeTextEditor.document.fileName).then(pkgMap => this.pkgList = pkgMap);

let setpkgslist = Promise.race([timeout(1000).then(() => this.pkgsList), importablePkgsPromise]);
```

Case when gopkgs runs under a second: 
Completion will always use the latest result

Case when gopkgs runs over a second: 
Completion will use cache. If its the first completion call, then no pkgs will be available which is fine, we can attribute this to cold start
When gopkgs eventually finishes, it will update the result, so that the next completion gets to use a more recent cache

Reply:
The reason I am suggesting the above is that with your current design 
- a user can still get stuck waiting for the very first call to completion
- a user can get stale data if gopkgs always takes over 1 second, and the next call from completion happens after 5 sec (the cache expiry time)

Reply:
Ahh... I see